pkg/gui/controllers/helpers/repos_helper.go

```diff
@@ -173,11 +173,6 @@ func (self *ReposHelper) DispatchSwitchTo(path string, errMsg string, contextKey
 			return err
 		}
 
-		// these two mutexes are used by our background goroutines (triggered via `self.goEvery`. We don't want to
-		// switch to a repo while one of these goroutines is in the process of updating something
-		self.c.Mutexes().SyncMutex.Lock()
-		defer self.c.Mutexes().SyncMutex.Unlock()
```

Comment:
Right, this could be a problem. But I agree that it's problem with or without the sync mutex, so we don't have to worry about it in this context. Maybe it could be solved somehow in that giant concurrency refactor that I want to do (and that I haven't made much progress on so far...).